src/projects/gt4o_us/quick-arcade/ArcadeRoot.ad

```diff
@@ -0,0 +1,533 @@
+//---------------------------------------------------------------------------------------
+// ArcadeRoot.ad
+//
+// Arcade mode Pre-race Menu: Main menu
+// 
+//---------------------------------------------------------------------------------------
+
+
+//---------------------------------------------------------------------------------------
+// module
+//---------------------------------------------------------------------------------------
+
+module QuickArcadeProject::ArcadeRoot
+{
+	//#include "../quick-share/icon.ad";
+	// Unsure what the deal is with this, the logic here is different??
+	// This decomp'd script uses a vertical transition with the grid list; the items slide from the bottom
+	// upward into place. The original script used the familiar horizontal transition where the list items slide from the right edge into place.
+	// icon.ad was included, but each script contains different logic from its icon.ad inclusion.
+	// It is probably best to axe this particular include, or, create separate script name for this include
+	// begin icon.ad included stuff
+	
+	function setFadeActor(widget, Velocity, Depth)
+	{
+		widget.actor = main::menu::MFadeActor(widget, 1.0);
+		widget.actor.velocity = Velocity;
+		widget.actor.out = 1;
+		widget.actor.over = Depth;
+		widget.actor.warp();
+		widget.actor.out = 0;
+	}
+
+	function rewindFadeActor(widget)
+	{
+		widget.actor.out = 1;
+		widget.actor.warp();
+		widget.actor.out = 0;
+	}
+
+	function setScrollActor(widget)
+	{
+		widget.actor = main::menu::MMoveActor(widget, 0.0, 0.0);
+		widget.actor.ratio = 0.2;
+		widget.actor.out = 1;
+		widget.actor.warp();
+	}
+
+	function setMoveActor(widget, x, y, r)
+	{
+		var actor = main::menu::MMoveActor(widget, widget.x + x, widget.y + y);
+		widget.actor = actor;
+		widget.actor.ratio = r;
+		widget.actor.out = 1;
+		widget.actor.warp();
+	}
+
+	function rewindMoveActor(widget)
+	{
+		widget.actor.out = 1;
+		widget.actor.warp();
+		widget.actor.out = 0;
+	}
+
+	function getIconTopIndex(context)
+	{
+		var num = icon_name_tbl.size;
+		var i = 0;
+	
+		while (i < num)
+		{
+			if (ArcadeRoot::Buttons[icon_name_tbl[i]].visible == 1)
+			{
+				return i;
+			}
+	
+			i++;
+		}
+	
+		return -1;
+	}
+
+	function setDefaultIconFocus(context)
+	{
+		var quick_work = main::menu::MQuickWork();
+		var idx = quick_work.cursorPosition;
+	
+		if (ArcadeRoot::Buttons[icon_name_tbl[idx]].visible != 1)
+		{
+			idx = getIconTopIndex(context);
+		}
+	
+		ArcadeRoot.setFocus(Buttons[icon_name_tbl[idx]]);
+	}
+
+	function getIconIndex(context, icon_name)
+	{
+		var idx = -1;
+		var num = icon_name_tbl.size;
+	
+		for (var i = 0; i < num; i++)
+		{
+			if (icon_name_tbl[i] == icon_name)
+			{
+				idx = i;
+				break;
+			}

```

Comment:
Dissassembly claims there's an extra redundant jump here, so it's probably a `if {} else {}` block

```js
if (icon_name_tbl[i] == icon_name)
{
    idx = i;
    break;
}
else
    ;
```